src/main/java/filter/SemanticException.java

```diff
@@ -0,0 +1,11 @@
+package filter;
+
+import filter.expression.QualifierType;
+
+@SuppressWarnings("serial")
+public class SemanticException extends FilterException {
+    public SemanticException(QualifierType type) {
+        super(String.format("Invalid input in filter for type \"%s\"", type));
```

Comment:
We don't have a lot of room in the status bar, so we might want to make the preamble shorter.

Also, this constructor could take another parameter for the details of the error. We should explain why the input causes a problem.